optuna_dashboard/static/components/HyperparameterImportances.tsx

```diff
@@ -0,0 +1,88 @@
+import * as plotly from "plotly.js-dist"
+import React, { FC, useEffect } from "react"
+import { getParamImportances } from "../apiClient"
+const plotDomId = "graph-hyperparameter-importances"
+
+// To match colors used by plot_param_importances in optuna.
+const plotlyColorsSequentialBlues = [
+  "rgb(247,251,255)",
+  "rgb(222,235,247)",
+  "rgb(198,219,239)",
+  "rgb(158,202,225)",
+  "rgb(107,174,214)",
+  "rgb(66,146,198)",
+  "rgb(33,113,181)",
+  "rgb(8,81,156)",
+  "rgb(8,48,107)",
+]
+
+const distributionColors = {
+  UniformDistribution: plotlyColorsSequentialBlues.slice(-1)[0],
+  LogUniformDistribution: plotlyColorsSequentialBlues.slice(-1)[0],
+  DiscreteUniformDistribution: plotlyColorsSequentialBlues.slice(-1)[0],
+  IntUniformDistribution: plotlyColorsSequentialBlues.slice(-2)[0],
+  IntLogUniformDistribution: plotlyColorsSequentialBlues.slice(-2)[0],
+  CategoricalDistribution: plotlyColorsSequentialBlues.slice(-4)[0],
+}
+
+export const HyperparameterImportances: FC<{
+  studyId: number
+}> = ({ studyId }) => {
+  useEffect(() => {
+    async function fetchAndPlotParamImportances(studyId: number) {
+      const paramsImportanceData = await getParamImportances(studyId)
+      plotParamImportances(paramsImportanceData)
+    }
+    fetchAndPlotParamImportances(studyId)
+  })
```

Comment:
https://user-images.githubusercontent.com/5564044/111020398-d5698800-8408-11eb-9a3d-15f89e455eaa.mp4


Please set `deps` to avoid sending API requests at every render.

```suggestion
  }, [studyId])
```

Reply:
Oh thanks! I missed that part, re-rendering could be unnecessary.

Reply:
I was just thinking sometimes `studyId` could not be a proper dependency as trials may get updated while the `studyId ` remains the same while parameter importance requires a re-evaluation.